app/projectsmodel.cpp

```diff
@@ -14,192 +8,582 @@
+    mProjects.clear();
+
+  if ( mModelType == ProjectModelTypes::LocalProjectsModel )
+  {
+    // Keep all local projects and ignore all not downloaded remote projects
+    for ( const auto &localProject : localProjects )
     {
-      projectFile.info = QString( created.toString() );
+      std::shared_ptr<Project> project = std::shared_ptr<Project>( new Project() );
+      project->local = std::unique_ptr<LocalProject>( new LocalProject( localProject ) );
+
+      MerginProject remoteEntry;
+      remoteEntry.projectName = project->local->projectName;
+      remoteEntry.projectNamespace = project->local->projectNamespace;
+
+      if ( merginProjects.contains( remoteEntry ) )
+      {
+        int i = merginProjects.indexOf( remoteEntry );
+        project->mergin = std::unique_ptr<MerginProject>( new MerginProject( merginProjects[i] ) );
+
+        if ( pendingProjects.contains( project->mergin->id() ) )
+        {
+          TransactionStatus projectTransaction = pendingProjects.value( project->mergin->id() );
+          project->mergin->progress = projectTransaction.transferedSize / projectTransaction.totalSize;
+          project->mergin->pending = true;
+        }
+        project->mergin->status = ProjectStatus::projectStatus( project );
+      }
+      else if ( project->local->localVersion > -1 )
+      {
+        // this is indeed a Mergin project, it has metadata folder in it
+        project->mergin = std::unique_ptr<MerginProject>( new MerginProject() );
+        project->mergin->projectName = project->local->projectName;
+        project->mergin->projectNamespace = project->local->projectNamespace;
+        project->mergin->status = ProjectStatus::projectStatus( project );
+      }
+
+      mProjects << project;
     }
-    else
+  }
+  else if ( mModelType != ProjectModelTypes::RecentProjectsModel )
+  {
+    // Keep all remote projects and ignore all non mergin projects from local projects
+    for ( const auto &remoteEntry : merginProjects )
     {
-      projectFile.info = project.qgisProjectError;
+      std::shared_ptr<Project> project = std::shared_ptr<Project>( new Project() );
+      project->mergin = std::unique_ptr<MerginProject>( new MerginProject( remoteEntry ) );
+
+      if ( pendingProjects.contains( project->mergin->id() ) )
+      {
+        TransactionStatus projectTransaction = pendingProjects.value( project->mergin->id() );
+        project->mergin->progress = projectTransaction.transferedSize / projectTransaction.totalSize;
+        project->mergin->pending = true;
+      }
+
+      // find downloaded projects
+      LocalProject localProject;
+      localProject.projectName = project->mergin->projectName;
+      localProject.projectNamespace = project->mergin->projectNamespace;
+
+      if ( localProjects.contains( localProject ) )
+      {
+        int ix = localProjects.indexOf( localProject );
+        project->local = std::unique_ptr<LocalProject>( new LocalProject( localProjects[ix] ) );
```

Comment:
impleement `clone`  method for all classes where you need to do something like this